pkg/server/proto/server.proto

```diff
@@ -5420,6 +5472,264 @@ message ProjectTemplate {
+  PaginationResponse pagination = 2;
+  uint64 total_count = 3;
+}
+
+// CreateAddOnDefinitionRequest is the request used to create an Add On
+// definition
+message CreateAddOnDefinitionRequest {
+  // Name of the Add On definition
+  string name = 1;
+
+  // Terraform No Code module used for provisioning the Add On
+  TerraformNocodeModule terraform_nocode_module = 2;
+
+  // Short description of the Add On
+  string short_summary = 3;
+
+  // Longer description of the Add On
+  string long_summary = 4;
+
+  // A templated README describing the Add On
+  bytes readme_markdown_template = 5;
+
+  // A list of descriptive tags for an Add On
+  repeated string tags = 6;
+
+  // The TF variable set to apply to the Add On's No Code workspace
+  repeated string tf_variable_set_ids = 7;
+
+  // The type of the Add On
+  AddOnType type = 8;
+}
+
+// CreateAddOnDefinitionResponse is the response containing the just-created
+// Add On definition
+message CreateAddOnDefinitionResponse {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// UpdateAddOnDefinitionRequest is the request used to update an existing
+// Add On definition
+message UpdateAddOnDefinitionRequest {
+  // Reference to an existing Add On
+  Ref.AddOnDefinition definition = 1;
+
+  // Name of the Add On definition
+  string name = 2;
+
+  // Terraform No Code module used for provisioning the Add On
+  TerraformNocodeModule terraform_nocode_module = 3;
+
+  // Short description of the Add On
+  string short_summary = 4;
+
+  // Longer description of the Add On
+  string long_summary = 5;
+
+  // A templated README describing the Add On
+  bytes readme_markdown_template = 6;
+
+  // A list of descriptive tags for an Add On
+  repeated string tags = 7;
+
+  // The TF variable set to apply to the Add On's No Code workspace
+  repeated string tf_variable_set_id = 8;
+
+  // The type of the Add On
+  AddOnType type = 9;
+}
+
+// UpdateAddOnDefinitionResponse is the response containing the just-updated
+// Add On definition
+message UpdateAddOnDefinitionResponse {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// DeleteAddOnDefinitionRequest is the request used to delete an existing Add
+// On definition
+message DeleteAddOnDefinitionRequest {
+  Ref.AddOnDefinition add_on_definition = 1;
+}
+
+// CreateAddOnRequest is the request used to add an Add On to a project
+message CreateAddOnRequest {
+  string name = 1;
+
+  // The project for which the Add On is to be created
+  Ref.Project project = 3;
+
+  // The Add On definition from which this Add On is to be created
+  Ref.AddOnDefinition definition = 4;
+
+  // Name of the Add On's Terraform workspace
+  string tf_workspace_name = 5;
+
+  // Rendered README markdown template for Add On
+  bytes readme_markdown = 6;
+
+  // A list of descriptive tags for an Add On
+  repeated string tags = 7;
+
+  // The type of the Add On
+  AddOnType type = 8;
+}
+
+enum AddOnType {
+  other = 0;
+  database = 1;
+  search = 2;
+  monitor = 3;
+  storage = 4;
```

Comment:
I believe it's standard to have enums be `UPPERCASE`, which is consistent with other parts of our `server.proto` file usage of enums

- https://protobuf.dev/programming-guides/style/#enums

Reply:
I uppercased all of these, however `OTHER` and `STORAGE` are already in use by [another part of the protobuf definition](https://github.com/hashicorp/waypoint/blob/344b5a4163a9c8490c5325865840c780f7176d8c/pkg/server/proto/server.proto#L1544-L1554). Because the strings are the exact same when uppercased, and they're in the same scope (`hashicorp.Waypoint`), they violate the [rules of enum in protobuf](https://github.com/protocolbuffers/protobuf/issues/5425). For that reason, I've now prefixed all of these with `ADDON_`.

Reply:
Good idea 👍